Lib/test/string_tests.py

```diff
@@ -682,6 +682,42 @@ def test_replace_overflow(self):
         self.checkraises(OverflowError, A2_16, "replace", "A", A2_16)
         self.checkraises(OverflowError, A2_16, "replace", "AA", A2_16+A2_16)
 
+    def test_cutprefix(self):
+        self.checkequal('am', 'spam', 'cutprefix', 'sp')
+        self.checkequal('spamspam', 'spamspamspam', 'cutprefix', 'spam')
+        self.checkequal('spam', 'spam', 'cutprefix', 'python')
```

Comment:
Are the below tests with "spider" and "spam and eggs" doing the same thing as this test or do they check for some other behavior? Same with cutsuffix test.

Reply:
I was trying to enforce that this method is from different from `a.lstrip(b)`.

In the "spam and eggs" case, I was trying to test that nothing breaks when the string is a prefix of the argument, such as a segfault, or if the implementation had compared `a[:min(len(a),len(b))] == b[:min(len(a),len(b))]`

Analogously with the cutsuffix tests.